frontend/packages/integration-tests-cypress/support/index.ts

```diff
@@ -40,9 +41,17 @@ after(() => {
 
 export const checkErrors = () =>
   cy.window().then((win) => {
-    if (win.windowError) {
-      throw new Error(`window/js runtime error detected: ${win.windowError}`);
-    }
+    const assertMsg =
+      win.windowError !== null
+        ? `window error detected: ${(win.windowError as PromiseRejectionEvent).reason ||
+            (win.windowError as Error).message} ${
+            (win.windowError as Error).stack
+              ? (win.windowError as Error).stack.replace(/\\n/g, '\n')
+              : ''
+          }`
```

Comment:
This is too much on one line. I'd pull this out into a helper. Something like

```ts
const formatWindowError = (windowError: Error | PromiseRejectionEvent) => {
  const { reason } = windowError as PromiseRejectionEvent;
  if (reason) {
    return reason;
  }
  const { message, stack } = windowError as Error;
  const formattedStack = stack?.replace(/\\n/g, '\n');
  return `${message} ${formattedStack}`;
};
```

Reply:
thanks, fixed.